### PR TITLE
Hyperlink to A+ Token Action

### LINKED ISSUE #509: Hyperlink to A+ Token Action
In Assignments view the label informing the user to set the A+ token could be turned into a hyperlink that when clicked would trigger the action. The priority for this is very low since this action is not expected to be performed very often. Also, these hyperlink-labels (both in Modules and Assignments views) could be styled more like a link (e.g. with the use of LinkLabel).

### PR DESCRIPTION
**Pull request by jaakkonarhi**
# Description of the PR

+ Closes #509 
+ You can't change the color of the text 

# Testing
<table>
  <tr>
    <th>unit</th>
    <th>integration</th>
    <th>e2e</th>
    <th>manual</th>
  </tr>
  <tr>
    <td>
      <ul>
        <li>- [ ] new <b>unit</b> tests created</li>
        <li>- [x] all <b>unit</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [ ] new <b>integration</b> tests created</li>
        <li>- [x] all <b>integration</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [ ] new <b>e2e</b> tests created</li>
        <li>- [x] all <b>e2e</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [x] <b>manual</b> testing went well</li>
      </ul>
    </td>
  </tr>
</table>  
  
# Have you updated the [TESTING.md](https://github.com/Aalto-LeTech/intellij-plugin/blob/master/TESTING.md) or other relevant documentation on _your_ branch?

- [ ] Yes.
- [ ] Not yet. I will do it next.
- [x] Not relevant.

### CODE REVIEW

**Review by stellatsv — CHANGES_REQUESTED** [review]
This works great! One note would be to move the listener class from `ExercisesView` to `APlusToolWindowFactory`. You can look at how this is done for the `ModulesView`.

File: src/main/java/fi/aalto/cs/apluscourses/ui/exercise/ExercisesView.java [hunk 1]
```diff
@@ -106,4 +114,16 @@ public JLabel getEmptyTextLabel() {
...
 
+  private class NoTokenMouseAdapter extends MouseAdapter {
+    @Override
+    public void mouseClicked(MouseEvent e) {
+      if (exerciseGroupsTree.isEmpty()
+          && exerciseGroupsTree.getEmptyText().getText().contains(
+              getText("ui.exercise.ExercisesView.setToken"))) {
+        DataContext context = DataManager.getInstance().getDataContext(e.getComponent());
+        ActionUtil.launch(APlusAuthenticationAction.ACTION_ID, context);
+      }
```
Comment (OlliKiljunen) [inline src/main/java/fi/aalto/cs/apluscourses/ui/exercise/ExercisesView.java:125]: Is there a need for consuming the event?

File: src/main/java/fi/aalto/cs/apluscourses/ui/exercise/ExercisesView.java [hunk 2]
```diff
@@ -67,12 +73,14 @@ public void viewModelChanged(@Nullable ExercisesTreeViewModel viewModel) {
         return;
       }
 
+      exerciseGroupsTree.removeMouseListener(mouseAdapter);
```
Comment (OlliKiljunen) [inline src/main/java/fi/aalto/cs/apluscourses/ui/exercise/ExercisesView.java:76]: Do we need to remove and re-add the listener always when the viewmodel changes? Can it just be there all the time?

**Review by OlliKiljunen — APPROVED** [review]
I don't see any remarkable problem in this implementation, and I don't see much reason to move the mouse event listener into the tool window factory.

Actually, how I would perhaps solved this is that the `TreeView` is given an action that it launches when it's clicked as empty. So, I would only have to override `processMouseEvent` in the TreeView and not work with the listeners. But this seems fair enough, too.

File: src/main/java/fi/aalto/cs/apluscourses/ui/exercise/ExercisesView.java (hunk 1 above)
Comment (jaakkonarhi) [reply src/main/java/fi/aalto/cs/apluscourses/ui/exercise/ExercisesView.java:124]: I'm not sure about this, but I don't think so

File: src/main/java/fi/aalto/cs/apluscourses/ui/exercise/ExercisesView.java (hunk 2 above)
Comment (jaakkonarhi) [reply src/main/java/fi/aalto/cs/apluscourses/ui/exercise/ExercisesView.java]: It can be there all the time

**Review by stellatsv — APPROVED** [review]

**Pull request merged**